### PR TITLE
fix up the dsat tests

### PR DESCRIPTION
**Pull request by tayritenour**
## Description
A bunch of changes to the autotuning interface broke the tests, this mends them and attempts to generalize a bit.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: harness/determined/pytorch/deepspeed/dsat/_dsat_search_method.py [hunk 1]
```diff
@@ -484,7 +484,7 @@ class BaseDSATSearchMethod(searcher.SearchMethod):
     needs to be written overwritten when subclassing (at a minimum).
     """
 
-    def __init__(self, args: argparse.Namespace, exp_config=Dict[str, Any]) -> None:
+    def __init__(self, args: argparse.Namespace, exp_config: Dict[str, Any]) -> None:
```
Comment (garrett361) [inline harness/determined/pytorch/deepspeed/dsat/_dsat_search_method.py:487]: Oops, good catch.

File: harness/tests/experiment/pytorch/test_deepspeed_autotuning.py [hunk 2]
```diff
@@ -242,7 +235,7 @@ def _append_events_for_op(self, op: searcher.Operation) -> None:
             self.events_queue.append(event)
 
         elif isinstance(op, searcher.Shutdown):
-            exp_state = bindings.experimentv1State.STATE_COMPLETED
+            exp_state = bindings.experimentv1State.COMPLETED
```
Comment (garrett361) [inline harness/tests/experiment/pytorch/test_deepspeed_autotuning.py:238]: what's up with the various enum changes? and isn't `STATE_COMPLETED` the right value here? I'm seeing
```python
class experimentv1State(enum.Enum):
    STATE_UNSPECIFIED = "STATE_UNSPECIFIED"
    STATE_ACTIVE = "STATE_ACTIVE"
    STATE_PAUSED = "STATE_PAUSED"
    STATE_STOPPING_COMPLETED = "STATE_STOPPING_COMPLETED"
    STATE_STOPPING_CANCELED = "STATE_STOPPING_CANCELED"
    STATE_STOPPING_ERROR = "STATE_STOPPING_ERROR"
    STATE_COMPLETED = "STATE_COMPLETED"
    STATE_CANCELED = "STATE_CANCELED"
    STATE_ERROR = "STATE_ERROR"
    STATE_DELETED = "STATE_DELETED"
    STATE_DELETING = "STATE_DELETING"
    STATE_DELETE_FAILED = "STATE_DELETE_FAILED"
    STATE_STOPPING_KILLED = "STATE_STOPPING_KILLED"
    STATE_QUEUED = "STATE_QUEUED"
    STATE_PULLING = "STATE_PULLING"
    STATE_STARTING = "STATE_STARTING"
    STATE_RUNNING = "STATE_RUNNING"
```

**Review by garrett361 — APPROVED** [review]
Just one substantive comment.

File: harness/tests/experiment/pytorch/test_deepspeed_autotuning.py (hunk 2 above)
Comment (tayritenour) [reply harness/tests/experiment/pytorch/test_deepspeed_autotuning.py:238]: They changed it here: https://github.com/determined-ai/determined/pull/6496 The bindings are the same but some of the names to python internal APIs are shortened now

File: harness/tests/experiment/pytorch/test_deepspeed_autotuning.py [hunk 3]
```diff
@@ -62,11 +41,25 @@ def _run_searcher(search_method: BaseDSATSearchMethod, all_metrics):
...
+            start_profile_step=_defaults.AUTOTUNING_ARG_DEFAULTS["start-profile-step"],
+            end_profile_step=_defaults.AUTOTUNING_ARG_DEFAULTS["end-profile-step"],
+            deepspeed_config=_defaults.AUTOTUNING_ARG_DEFAULTS["deepspeed-config"],
+            metric=_defaults.AUTOTUNING_ARG_DEFAULTS["metric"],
+            # NONE TYPES
+            max_slots=None,
+            early_stopping=None,
         )
+        searcher_dir = pathlib.Path(searcher_dir)
+        search_method = search_method(args=args, exp_config=submitted_config_dict)
```
[the remark this answers is not in the document]
Comment (tayritenour) [reply harness/tests/experiment/pytorch/test_deepspeed_autotuning.py:62]: Yup, I can make that change. And I agree. I like this direction a lot more in terms of making the UX a lot more pleasant.

**Pull request merged**